src/tree/registries/azure/AzureAccountTreeItem.ts

```diff
@@ -28,4 +30,33 @@ export class AzureAccountTreeItem extends AzureAccountTreeItemBase implements IR
     public createSubscriptionTreeItem(subContext: ISubscriptionContext): SubscriptionTreeItem {
         return new SubscriptionTreeItem(this, subContext);
     }
+
+    public async loadMoreChildrenImpl(_clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]> {
+        const treeItems: AzExtTreeItem[] = await super.loadMoreChildrenImpl(_clearCache, context);
+
+        if (treeItems.length === 1 && treeItems[0].commandId === 'extension.open') {
+            // tslint:disable-next-line: no-floating-promises
+            this.refreshNodeWhenAzureExtensionIsInstalled();
```

Comment:
This could get called multiple times, if the user manually refreshes this node, which means multiple waiters for the installation.  Moreover, if the user manually refreshes the parent node (e.g. the root tree), new instances of this class will be created and this function could then be called multiple times across several instances.  That means it's not enough to track waiting at the instance level; it needs to be statically tracked so there is at most a single waiter.

Reply:
Now using event